cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/RunRecordMonitorService.java

```diff
@@ -48,6 +48,12 @@
 public class RunRecordMonitorService extends AbstractScheduledService {
   private static final Logger LOG = LoggerFactory.getLogger(RunRecordMonitorService.class);
 
+  /**
+   * Start calculating precise number of running programs after {@link this#maxConcurrentRuns}
+   * has reached this fraction.
+   */
+  private static final float FRACTION_FOR_PRECISE_RUNNING_COUNTING = .9f;
```

Comment:
why don't we do lazy calculating ?

since the imprecise running count must be greater than actual running count, we can do the precise computing once the limit exceeds , WDYT ?